src/app/state/rest/useApiRequest.ts

```diff
@@ -1,41 +1,28 @@
-import axios, { AxiosError, AxiosResponse, CancelTokenSource, Method } from "axios"
-import { useState, useCallback, useEffect, useRef } from "react"
-import slashSandwich from "slash-sandwich"
+import axios, { AxiosError, AxiosResponse, Method } from "axios"
+import { useState, useCallback, useEffect, useContext } from "react"
 
 import { getToken } from "../auth/tokenStore"
-import cache from "./CacheStore"
 import { useFlashMessages } from "../flashMessages/useFlashMessages"
+import { ApiCacheContext } from "./ApiCacheProvider"
+import useIsMounted from "../../features/shared/hooks/useIsMounted/useIsMounted"
 
 type Config = {
-  group: string
   url: string
 }
 
-/**
- * Setup axios-client:
- * - Configures a cache adapter
- */
-const api = axios.create({
-  adapter: cache.adapter
-})
+const pending: Record<string, Promise<AxiosResponse<any>>> = {}
```

Comment:
`pending` is shared between hooks. So should be placed inside a context.

Reply:
That would only add complexity in my opinion.
What extra value would it bring?

Reply:
Otherwise if two components request the same data it could be fetched twice

Reply:
No it's not :-).

The pending object is defined outside the hook, and thus wouldn't be re-initialized every render.
By defining it outside the hook it's shared between hooks.

Reply:
On line 51 when the promise gets resolved it deletes itself from the shared `pending` object. This will happen before any possible additional resolving function (from another component on the same promise) is called.

Reply:
It waits for the promise to resolve on line 50